stakingModel/test/BasicCalculation.test.js

```diff
@@ -12,144 +12,127 @@ contract("InterestDistribution - Basic calculations", ([user1]) => {
     
   });
 
-  it("Should return correct Avg Yield Rate Per Token", async () => {
+  it("Should return correct cumulative Yield Rate", async () => {
 
-      // 1st stake will not affect AvgYieldRatePerToken and GlobalYieldPerToken because globalTotalStaked and totalStaked both are 0.
+      // 1st stake will not affect GDYieldRate and GlobalYieldPerToken because globalTotalStaked is 0.
       await interestDistribution
-        .stakeCalculation(user1, web3.utils.toWei("20", "ether"), 0, web3.utils.toWei("1", "ether"), web3.utils.toWei("1", "ether"))
+        .stake(user1, web3.utils.toWei("20", "ether"), 0, 0, 0)
         .catch(e => e);
 
-
       /**
       * Formula:
-      * GlobalYieldPerToken = GlobalYieldPerToken(P) + [(IntrestTokenRate - LastTokenRate) x IntrestTokenHoldings]/GlobalTotalStake
-      * AvgYieldRatePerToken = [(AvgYieldRatePerToken(P) x TotalStaked) + (AccumulatedYieldPerToken(P) x Staking x (1-%Donation))]/TotalStaked
+      * GlobalYieldPerToken = GlobalYieldPerToken(P) + GDEarnedInterest/GlobalTotalEffectiveStake.
+      * GDYieldRate = [GDYieldRate(P) + (AccumulatedYieldPerToken(P) x EffectiveStake)]
       * consider :
-      * IntrestTokenRate = 1.02
-      * LastTokenRate = 1
-      * IntrestTokenHoldings = 30 x 1e18
-      * GlobalTotalStake = 20 x 1e18
-      * AccumulatedYieldPerToken(P) = 3
-      * AvgYieldRatePerToken(P) = 0
+      * GlobalTotalEffectiveStake = 20 x 1e18
+      * AccumulatedYieldPerToken(P) = 0
+      * GDYieldRate(P) = 0
       * Staking = 10 x 1e18
       * Donation% = 30%
-      * TotalStaked = 20 x 1e18
-      * output = 3
+      * GDEarnedInterest = 100 x 1e18
+      * GDEarnedInterestEarned = 100 x 1e18
       */
 
       await interestDistribution
-        .stakeCalculation(user1, web3.utils.toWei("10", "ether"), 30, web3.utils.toWei("1.02", "ether"), web3.utils.toWei("30", "ether"))
+        .stake(user1, web3.utils.toWei("10", "ether"), 30, web3.utils.toWei("100", "ether"), web3.utils.toWei("100", "ether"))
```

Comment:
use numbers that make sense
G$ is in 2 decimals points. so G$ earned interest will not be 100 ether. the earned interest in token should also have logical numbers relative to G$ value

Reply:
Done